index.html

```diff
@@ -2083,34 +2084,59 @@ <h3 id="applying">
           </li>
           <li>If <var>src</var> is <code>undefined</code>, or if the result of
           running <a>Is origin potentially trustworthy</a>
-          with the origin of <var>src</var> is <code>Not Trusted</code>, return
-          <code>undefined</code>.
+          with the origin of <var>src</var> is <code>Not Trusted</code>, abort
+          these steps and return <code>undefined</code>.
           </li>
           <li>Otherwise, let <var>registration</var> be an object with
-          properties <code>src</code> and <code>scope</code>. All properties
-          initially set to <code>undefined</code>.
+          properties <code>src</code>, <code>scope</code>, <code>type</code> and
+          <code>use_cache</code>. All properties initially set to
+          <code>undefined</code>.
           </li>
           <li>Set <var>registration</var>'s <code>src</code> property to be
           <var>src</var>.
           </li>
-          <li>Let <var>type</var> be the result of running the <a>steps
+          <li>Let <var>scope</var> be the result of running the <a>steps
           for processing the <code>scope</code> member of a service worker</a>
           passing <var>unprocessed registration</var>.
           </li>
           <li>If <var>scope</var> is not <code>undefined</code>,
           set <var>registration</var>'s <code>scope</code>
           property to be <var>scope</var>.
           </li>
+          <li>Let <var>type</var> be the result of running the <a>steps
+          for processing the <code>type</code> member of a service worker</a>
+          passing <var>unprocessed registration</var>.
+          </li>
+          <li>If <var>type</var> is not <code>undefined</code>
+          set <var>registration</var>'s <code>type</code>
+          property to be <var>type</var>.
+          </li>
+          <li>If <var>type</var> is <code>undefined</code> abort these steps and
+          return <code>undefined</code>.
```

Comment:
Service workers expect a default value for type member. If type is undefined here, please set the default value to "classic" when passing it to Start Register.